### ISSUE TITLE
Is there a way to automatically reload modifications to main.py?

### ISSUE THREAD

**Issue by ilpersi** [GoogleCloudPlatform/functions-framework-python#6]
I know that in standard Flask you can do this in the following way:
```
$ export FLASK_APP=main.py
$ export FLASK_ENV=development
$ flask run
```

I've tried also with functions-framework with no luck: is there a possible workaround?
Thank you.

**Comment by di** [comment]
Yes, you can use the `--debug TRUE` command line flag or set the `DEBUG` environment variable to `TRUE`.

**Comment by ilpersi** [comment]
I saw the `--debug TRUE` switch in #5 and used it to debug my functions; it does not automatically reload the main.py after I modify it: I have to stop function-framework and start it again

**Comment by di** [comment]
Can you give us some more details about how you're trying to run the `functions-framework`? The following works for me

```
$ cat main.py
def hello(request):
    return "Hello world!"

$ functions-framework --target hello --debug TRUE
 * Serving Flask app "hello" (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: on
 * Running on http://0.0.0.0:8080/ (Press CTRL+C to quit)
 * Restarting with fsevents reloader
 * Debugger is active!
 * Debugger PIN: 137-358-281

<wrote to main.py>

 * Detected change in '/Users/dustiningram/main.py', reloading
 * Detected change in '/Users/dustiningram/main.py', reloading
 * Restarting with fsevents reloader
 * Debugger is active!
 * Debugger PIN: 137-358-281

$ flask --version
Python 3.7.4
Flask 1.1.1
Werkzeug 0.16.0
```

**Comment by ilpersi** [comment]
Hi Dustin, here is a video of what I am doing: https://drive.google.com/file/d/1BJx4Bmc2YN4sgLTKDRSbqo-EdX1TZomj/preview

Is there any other detail that can help you?

**Comment by di** [comment]
Hi @ilpersi, thanks for the very detailed video! One difference is that Flask/Werkzeug is using the "stat" reloader on your machine, but I'm using "fsevents" which comes from the [watchdog](https://pypi.org/project/watchdog/) package.

It seems like what Flask uses varies from one platform to another. Could you try installing watchdog in your environment and see if that helps?

Also if you could include the output of `flask --version` that'd be great. Thanks!

**Comment by ilpersi** [comment]
@di installing watchdog solves the issue! 🔝 👍 

Here is my `flask --version`
```
Python 3.7.4
Flask 1.1.1
Werkzeug 0.16.0
```

I think that Flask version is also something that deserves more documentation: to stick with the [cloud functions runtime](https://cloud.google.com/functions/docs/writing/http#http_frameworks) I originally installed 1.0.2. and I updated to 1.1.1 after you posted your `flask --version` as an attempt to fix the issue.

The video in the previous comment is shot with 1.1.1 and installing watchdog fixes the issue both with 1.0.2 and 1.1.1 (I tested them both).

Thank you very much.
